inference-engine/tests/functional/plugin/cpu/subgraph_tests/src/fuse_eltwise_1st_port.cpp

```diff
@@ -0,0 +1,166 @@
+        std::string targetDevice;
+        bool connected_on_1st_port;
+        std::tie(inputShape, parent_precision, child_precision, eltwiseOpTypes, connected_on_1st_port, targetDevice) = obj.param;
+        std::ostringstream results;
+        results << "IS" << CommonTestUtils::vec2str(inputShape) << "_";
+        results << "ParPRC" << "=" << parent_precision << "_";
+        results << "ChildPRC" << "=" << child_precision << "_";
+        for (int i = 0; i < eltwiseOpTypes.size(); i++) {
+            results << "Op" << std::to_string(i) << "=" << eltwiseOpTypes[i] << "_";
+        }
+        results << "ConnectedOnPort" << "=" << (connected_on_1st_port ? 1 : 0) << "_";
+        results << "targetDevice=" << targetDevice;
+        return results.str();
+    }
+
+protected:
+    void SetUp() {
+        threshold = 0.1f;
+
+        Shape inputShape;
+        element::Type parent_precision, child_precision;
+        std::vector<EltwiseTypes> eltwiseOpTypes;
+        bool connected_on_1st_port;
+        std::tie(inputShape, parent_precision, child_precision, eltwiseOpTypes, connected_on_1st_port, targetDevice) = this->GetParam();
+
+        ASSERT_EQ(eltwiseOpTypes.size(), 2) << "Only two eltwise types are accepted";
+
+        auto param_parent = std::make_shared<opset6::Parameter>(parent_precision, inputShape);
+        auto param_child = std::make_shared<opset6::Parameter>(child_precision, inputShape);
+
+        std::vector<float> Input1Data2(1);
+        auto const_parent = ngraph::builder::makeConstant(parent_precision, Shape{1}, Input1Data2, true);
+        auto parent = ngraph::builder::makeEltwise(const_parent, param_parent, eltwiseOpTypes[0]);
+
+        std::shared_ptr<ngraph::Node> child;
+        if (connected_on_1st_port)
+            child = ngraph::builder::makeEltwiseRelaxed(param_child, parent, eltwiseOpTypes[1]);
+        else
+            child = ngraph::builder::makeEltwiseRelaxed(parent, param_child, eltwiseOpTypes[1]);
+
+        std::vector<std::shared_ptr<ngraph::Node>> nodes {parent, child};
+        for (auto node : nodes) {
+            ASSERT_EQ(node->get_input_size(), 2) <<
+                                                 "This test supports eltwise ops only with two inputs.";
+        }
+        ngraph::ResultVector results{std::make_shared<ngraph::opset6::Result>(child)};
+        ngraph::ParameterVector params{param_parent, param_child};
+
+        function = std::make_shared<ngraph::Function>(results, params, "FuseOnSecondPort");
+        #define cover_precisions(PARENT_PRECISION, CHILD_PRECISION) \
+        case element::CHILD_PRECISION: \
+            additionalPasses.push_back(std::make_shared<ngraph::pass::ConvertPrecision<element::PARENT_PRECISION, element::CHILD_PRECISION>>());\
+            break
+
+        switch (parent_precision) {
+            case element::i32: {
+                switch (child_precision) {
+                    cover_precisions(i32, f32);
```

Comment:
Why cannot we just create fully fp32 subgraph? Isn't it covers the code path?

Reply:
Ok. Seems like it is explained in the comments. Still looks like overkill from test purposes. Lets discuss offline.